File: web/src/types/market.ts
export type MarketCoin = {
  id: string;
  name: string;
  symbol: string;
  imageUrl: string | null;
  rank: number | null;
  price: number | null;
  change1h: number | null;
  change24h: number | null;
  change7d: number | null;
  marketCap: number | null;
  volume24h: number | null;
  sparkline: number[];
};

export type MarketStat = {
  label: string;
  value: string;
  helper: string;
  change: number | null;
};

File: api/app/schemas/market.py
from app.schemas.base import ApiModel


class MarketCoin(ApiModel):
    id: str
    name: str
    symbol: str
    rank: int
    price: float
    change_1h: float
    change_24h: float
    change_7d: float
    market_cap: float
    volume_24h: float
    sparkline: list[float]


class MarketStat(ApiModel):
    label: str
    value: str
    helper: str
    change: float | None = None
